### ISSUE TITLE
Array of structs

### ISSUE THREAD

**Issue by mesca** [csound/csound#1588]
It is now (7.0) possible to create arrays of structs with the following syntax: 
```
struct MyType a:i, b:i
myarr:MyType[] init 10
```

Unfortunately, it is currently not possible to access struct members directly with something like:
```
myarr[0].a = 42 ; this fails
```

There is a workaround, though. One has to create a temporary variable:
```
tmp:MyType = myarr[0]
tmp.a = 42
myarr[0] = tmp
```

Note that we need the same trick for reading a struct member value from an array:
```
 print(myarr[0].a) ; this fails too
```

This is neither efficient nor elegant. This is a [known problem](https://github.com/csound/csound/blob/22ee858cd5a03974222f3f16252cfa9daaf57e81/tests/commandline/structs/test_struct_arrays.csd#L40), but I am opening this issue so it can be tracked.

**Comment by hlolli** [comment]
sounds like something I'd want to see fixed. Going to attempt to implement this @kunstmusik

**Comment by kunstmusik** [comment]
Agreed this is a problem and should be optimized. It's been ages since I
looked at this so feel free to take a stab at it @hlolli or anyone else.
(Need to sort out dealing with pass-by-ref vs. pass-by-value)
